File: eclipse/02-Exercicio/src/br/com/fiap/bean/ContaPoupanca.java
package br.com.fiap.bean;

import java.util.Calendar;

import br.com.fiap.exception.SaldoInsuficienteException;

public class ContaPoupanca extends Conta implements ContaInvestimento{
	
	public static final float RENDIMENTO = 0.05f; // "f" no final para definir o tipo float.
	private float taxa;
	
	public ContaPoupanca(int agencia, int numero, Calendar dataAbertura, double saldo, float taxa) {
		super(agencia, numero, dataAbertura, saldo);
		this.taxa = taxa;
	}
	
	public double calculaRetornoInvestimento() {
		return RENDIMENTO * saldo;
	}

	@Override
	public void depositar(double valor) {
		saldo += valor;
	}

	@Override
	public void retirar(double valor) throws SaldoInsuficienteException {
		if(saldo < valor + taxa) {
			throw new SaldoInsuficienteException("Saldo insuficiente.");
		}
		saldo -= valor + taxa;
	}
}

File: .NET/aula2/Fiap.Bank/Models/ContaPoupanca.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Fiap.Bank.Models
{
    internal class ContaPoupanca : Conta, IContaInvestimento
    {
        decimal Taxa;
        decimal _rendimento;

        public ContaPoupanca(int agencia, int numero, DateTime dataAbertura, decimal saldo) : base(agencia, numero, dataAbertura, saldo)
        {
            Taxa = 2;
            _rendimento = 2;
        }

        public override void Depositar(decimal valor)
        {
            this.Saldo += valor;
        }

        public override void Retirar(decimal valor)
        {
            if(this.Saldo > (valor * this.Taxa))
            {
                this.Saldo -= (valor * this.Taxa);
                Console.WriteLine("Saldo retirado no valor: R$", valor);
            } else
            {
                Console.WriteLine("Saldo indisponivel");
            }
        }

        public decimal CalculaRetornoInvestimento()
        {
            return (this.Saldo * this._rendimento);
        }
    }
}
